### PR TITLE
HUM-313 Redirect STDERR/STDOUT when logging to files

### PR DESCRIPTION
**Pull request by creiht**

### CODE REVIEW

File: common/srv/server.go [hunk 1]
```diff
@@ -203,15 +203,31 @@ func SetupLogger(prefix string, atomicLevel *zap.AtomicLevel, flags *flag.FlagSe
...
+		f, err := os.OpenFile(lFlag.Value.(flag.Getter).Get().(string), os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to open file for logger: %v", err)
+		}
+		err = syscall.Dup2(int(f.Fd()), int(os.Stdout.Fd()))
+		if err != nil {
+			return nil, errors.New("Unable to redirect STDOUT")
+		}
+		lowPrioFile = zap.CombineWriteSyncers(f)
+		//lowPrioFile, _, openerr = zap.Open(lFlag.Value.(flag.Getter).Get().(string))
```
Comment (corystone) [inline common/srv/server.go]: Nuke this one. :)

File: common/srv/server.go [hunk 2]
```diff
@@ -203,15 +203,31 @@ func SetupLogger(prefix string, atomicLevel *zap.AtomicLevel, flags *flag.FlagSe
...
+		if err != nil {
+			return nil, fmt.Errorf("Unable to open file for logger: %v", err)
+		}
+		err = syscall.Dup2(int(f.Fd()), int(os.Stderr.Fd()))
+		if err != nil {
+			return nil, errors.New("Unable to redirect STDERR")
+		}
+		highPrioFile = zap.CombineWriteSyncers(f)
+
+		//highPrioFile, _, openerr = zap.Open(eFlag.Value.(flag.Getter).Get().(string))
```
Comment (corystone) [inline common/srv/server.go]: And this one. :)

File: common/srv/server.go [hunk 3]
```diff
@@ -203,15 +203,31 @@ func SetupLogger(prefix string, atomicLevel *zap.AtomicLevel, flags *flag.FlagSe
...
-		lowPrioFile, _, openerr = zap.Open(lFlag.Value.(flag.Getter).Get().(string))
+		f, err := os.OpenFile(lFlag.Value.(flag.Getter).Get().(string), os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to open file for logger: %v", err)
+		}
+		err = syscall.Dup2(int(f.Fd()), int(os.Stdout.Fd()))
+		if err != nil {
+			return nil, errors.New("Unable to redirect STDOUT")
+		}
+		lowPrioFile = zap.CombineWriteSyncers(f)
```
Comment (corystone) [inline common/srv/server.go]: I'm not sure this is actually necessary is it?

Comment (creiht) [reply common/srv/server.go]: I put this one in, that way if someone does some debugging with fmt.Println, they will show up in the logs.

**Comment by corystone** [comment]
LGTM! Panics work both ways.

**Pull request merged**